### ISSUE TITLE
Read a barcode without Bitmap .Net Core

### ISSUE THREAD

**Issue by xav-stargate** [micjahn/ZXing.Net#92]
Hey !

How can I read a barcode/qrcode in .Net Core without the class Bitmap ?
I try this, but I have this error : "Source must be non-null", the file exist ! and bytes is full !

byte[] bytes = File.ReadAllBytes(@"C:\Users\Xavier\Pictures\qrimage.png");
Result barcodeResult = barcodeReader.Decode(bytes);

:-/

**Comment by micjahn** [comment]
You can't do it that way. If you want to use a specific image format (like PNG in your example) you have to decode the image format first.
If you want to use the method "Decode" with only the byte array argument you have to declare a delegate in the constructor of the barcode reader which converts the byte array data to a luminance source object.

```
var barcodeReader = new BarcodeReader(
   null, 
   bytearray => 
      { 
         [...] your PNG conversion routine here which extracts the raw bytes, width, height and the format of the raw bytes from the byte array [...]; 
         return new RGBLuminanceSource(rawBytes, width, height, format);
      }, 
   null);
...
```

Alternativly you should try one of the differents bindings to other image manipulation libraries which are available for .Net Standard / .Net Core:
[https://www.nuget.org/packages?q=zxing.net.bindings](https://www.nuget.org/packages?q=zxing.net.bindings)
They contain specialized barcode reader classes which are consuming different image formats and do the conversion for you.

For example if you select the [https://www.nuget.org/packages/ZXing.Net.Bindings.CoreCompat.System.Drawing](https://www.nuget.org/packages/ZXing.Net.Bindings.CoreCompat.System.Drawing)
you can write the following:
```
var coreCompatReader = new ZXing.CoreCompat.System.Drawing.BarcodeReader();
using (var coreCompatImage = (System.Drawing.Bitmap) System.Drawing.Bitmap.FromFile(@"C:\Users\Xavier\Pictures\qrimage.png"))
{
   var coreCompatResult = coreCompatReader.Decode(coreCompatImage);
}
```

**Comment by xav-stargate** [comment]
Thank you it’s work ! :)

**Comment by a-a-k** [comment]
> `return new RGBLuminanceSource(rawBytes, width, height, format);`

@micjahn how could I retrieve height and width of image if it's type is unknown?

**Comment by JoseFrankGeraldino** [comment]
Currently I'm writing an app in blazor /wasm and using the System.Drawing Assembly is not an option because it still is not compatible with blazor/wasm. 
I have all the image data except the "format" that RGBLuminanceSource spects.

**Comment by JoseFrankGeraldino** [comment]
It seem that there are no plans for supporting System.Drawing in Blazor/Wasm
https://github.com/mono/mono/issues/11461

**Comment by denisbredikhin** [comment]
@JoseFrankGeraldino did you succeed with using ZXing.Net with Blazor WASM? I wanted to do the same and found that there are no ready examples or libraries for this...

**Comment by JoseGeraldino** [comment]
> @JoseFrankGeraldino did you succeed with using ZXing.Net with Blazor WASM? I wanted to do the same and found that there are no ready examples or libraries for this...

I gave up. I tried every available method to read the qr code. The last thing that I wanted to try was to import another wasm library (without using javascript) that reads qr codes, but I don't even know where to start. I think that that would be a great feature for Blazor, to import another wasm libraries. Hope that they have that feature in their roadmap.
